examples/finetune_resnet/components.py

```diff
@@ -0,0 +1,99 @@
+import os
+from collections import defaultdict
+from pickletools import optimize
+from typing import Any, Dict, List, Tuple
+
+import torch
+import torch.nn as nn
+from torch.optim import Adam
+from torchvision import datasets, models, transforms
+
+from tango import Step
+from tango.integrations.torch import DataCollator, Model, Optimizer
+
+# Register the Adam optimizer as an `Optimizer` so we can use it in the train step.
+Optimizer.register("torch_adam")(Adam)
+
+# Wrapper class around the pre-trained ResNet-18 model that modifies the final layer.
+@Model.register("resnet_ft")
+class ResNetWrapper(Model):
+
+    def __init__(self, num_classes: int, feature_extract: bool, use_pretrained: bool):
+        super().__init__()
+        self.model_ft = models.resnet18(pretrained=use_pretrained)
+        self.set_parameter_requires_grad(self.model_ft, feature_extract)
+        num_features = self.model_ft.fc.in_features
+        self.model_ft.fc = nn.Linear(num_features, num_classes)
+        self.loss_fn = nn.CrossEntropyLoss()
+
+    def set_parameter_requires_grad(self, model: models, feature_extracting: bool):
+        if feature_extracting:
+            for param in model.parameters():
+                param.requires_grad = False
+
+    def forward(self, image: torch.Tensor, label: torch.Tensor) -> Dict[str, torch.Tensor]:
+        output = self.model_ft(image)  # tensor
+        loss = self.loss_fn(output, label)
+        return {"loss": loss}
+
+# Custom data collator for images, that takes in a batch of images and labels and
+# reformats the data so that it is suitable for the model.
+@DataCollator.register("image_collator")
+class ImageCollator(DataCollator[Tuple[torch.Tensor]]):
+    def __call__(self, batch: List[Tuple[torch.Tensor]]) -> Dict[str, Any]:
```

Comment:
I think this should be:
```suggestion
    def __call__(self, batch: List[Tuple[torch.Tensor, int]]) -> Dict[str, Any]:
```

Reply:
yup makes sense, changing it